pootle/apps/pootle_store/views.py

```diff
@@ -321,16 +321,32 @@ def get_units(request):
         When the `initial` GET parameter is present, a sorted list of
         the result set ids will be returned too.
     """
-    pootle_path = request.GET.get('path', None)
+    category = request.GET.get('category')
+    checks = request.GET.get('checks')
+    chunk_size = request.GET.get('count')
+    is_initial_request = request.GET.get('initial', False)
+    modified_since = request.GET.get('modified-since')
+    month = request.GET.get('month')
+    pootle_path = request.GET.get('path')
+    search = request.GET.get('search')
+    sfields = request.GET.get('sfields')
+    sort_by_param = request.GET.get('sort')
+    unit_filter = request.GET.get('filter')
+    uids_param = filter(None, request.GET.get('uids', '').split(u','))
+    username = request.GET.get('user')
+
     if pootle_path is None:
         raise Http400(_('Arguments missing.'))
     elif len(pootle_path) > 2048:
         raise Http400(_('Path too long.'))
 
-    User = get_user_model()
-    request.profile = User.get(request.user)
-    limit = request.profile.get_unit_rows()
+    if request.user.is_anonymous():
+        # override the anon user
+        request.user = get_user_model().objects.get_nobody_user()
```

Comment:
> - it does an additional lookup for every user for almost no reason

If this is about the extra query on `get_step_query` it is now gone that you copied its code here. So why drop `User.get`?


> - removing request.profile

I agree with this.